transport/p2p/src/behavior/discovery.rs

```diff
@@ -236,14 +240,28 @@ impl NetworkBehaviour for Behaviour {
                     if peer != self.me {
                         tracing::debug!(%peer, addresses = ?&multiaddresses, "Announcement");
 
-                        for multiaddress in &multiaddresses {
+                        // Filter out private addresses before adding to pending events and peer store
+                        let public_addresses: Vec<_> = multiaddresses.iter()
```

Comment:
Would it be better to collect this to a `HashSet` instead so that it will deduplicate?

Reply:
Are multiaddresses Hashable?

Reply:
Yes: https://docs.rs/multiaddr/latest/multiaddr/struct.Multiaddr.html#impl-Hash-for-Multiaddr

Reply:
Done.